isisd/isis_misc.c

```diff
@@ -467,9 +468,11 @@ const char *print_sys_hostname(const uint8_t *sysid)
 	if (isis && !CHECK_FLAG(im->options, F_ISIS_UNIT_TEST))
 		return cmd_hostname_get();
 
-	dyn = dynhn_find_by_id(sysid);
-	if (dyn)
-		return dyn->hostname;
+	for (ALL_LIST_ELEMENTS_RO(im->isis, node, isis)) {
```

Comment:
Shouldn't we add an `isis` parameter instead of doing this?

Reply:
Ideally yes. But this function is called in a lot of places where the pointer to `isis` is not directly accessible.
In this PR I am focused on fixing severe problems - UAF and memory leaks, so I decided to keep things simple and just keep the old behavior here without reworking a lot of code. This function is only used for logging so I think it's fine.

Reply:
Fair enough!